src/modals/zaken/ZaakForm.vue

```diff
@@ -0,0 +1,239 @@
+				<NcNoteCard v-if="!success" type="error">
+					<p>Zaak niet succesvol {{ zaak.id ? 'aangepast' : 'aangemaakt' }}</p>
+				</NcNoteCard>
+				<NcNoteCard v-if="error" type="error">
+					<p>{{ error }}</p>
+				</NcNoteCard>
+			</div>
+
+			<div v-if="success === null" class="form-group">
+				<NcTextField :disabled="zaakLoading"
+					label="Identificatie"
+					maxlength="255"
+					:value.sync="zaak.identificatie"
+					required />
+				<NcTextField :disabled="zaakLoading"
+					label="Omschrijving"
+					maxlength="255"
+					:value.sync="zaak.omschrijving" />
+				<NcTextField :disabled="zaakLoading"
+					label="Bronorganisatie"
+					maxlength="9"
+					:value.sync="zaak.bronorganisatie"
+					required />
+				<NcTextField :disabled="zaakLoading"
+					label="VerantwoordelijkeOrganisatie"
+					maxlength="9"
+					:value.sync="zaak.verantwoordelijkeOrganisatie"
+					required />
+				<NcTextField :disabled="zaakLoading"
+					label="Startdatum"
+					maxlength="9"
+					:value.sync="zaak.startdatum"
+					required />
+				<NcSelect v-bind="zaakType"
+					v-model="zaakType.value"
+					input-label="Zaaktype"
+					:loading="zaakTypeLoading"
+					:disabled="zaakLoading || zaakTypeLoading"
+					required />
+				<NcSelect v-bind="archiefstatus"
+					v-model="archiefstatus.value"
+					input-label="Archiefstatus"
+					:disabled="zaakLoading"
+					required />
+				<NcTextField :disabled="zaakLoading"
+					label="Registratiedatum"
+					maxlength="255"
+					:value.sync="zaak.registratiedatum" />
+				<NcTextArea :disabled="zaakLoading"
+					label="Toelichting"
+					:value.sync="zaak.toelichting" />
+			</div>
+
+			<NcButton v-if="success === null"
+				:disabled="loading || !zaak.identificatie || !zaakTypeLoading || !zaak.bronorganisatie || !zaak.verantwoordelijkeOrganisatie || !zaak.startdatum"
+				type="primary"
+				@click="saveZaak()">
+				<template #icon>
+					<NcLoadingIcon v-if="loading" :size="20" />
+					<ContentSaveOutline v-else-if="!loading && zaak.id" :size="20" />
+					<Plus v-else-if="!loading && !zaak.id" :size="20" />
+				</template>
+				{{ zaak.id ? 'Opslaan' : 'Aanmaken' }}
+			</NcButton>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import {
+	NcModal,
+	NcNoteCard,
+	NcButton,
+	NcTextField,
+	NcSelect,
+	NcTextArea,
+	NcLoadingIcon,
+} from '@nextcloud/vue'
+
+// icons
+import Plus from 'vue-material-design-icons/Plus.vue'
+import ContentSaveOutline from 'vue-material-design-icons/ContentSaveOutline.vue'
+
+// entities
+import { Zaak } from '../../entities/index.js'
+
+export default {
+	name: 'ZaakForm',
+	components: {
+		NcModal,
+		NcTextField,
+		NcButton,
+		NcSelect,
+		NcTextArea,
+	},
+	data() {
+		return {
+			zaak: {
+				identificatie: '',
+				omschrijving: '',
+				zaaktype: {},
+				registratiedatum: '',
+				toelichting: '',
+				bronorganisatie: '',
+				verantwoordelijkeOrganisatie: '',
+				startdatum: '',
+				archiefstatus: '',
+			},
+			loading: false,
+			success: false,
```

Comment:
The variable `success` should be `null`, right now we will get an error card with the message "Zaak niet succesvol aangemaakt" which is blocking the use of the form.

Reply:
There should also be a variable named `error`
`error: null,`

Reply:
There should also be a variable named `zaakLoading`
`zaakLoading: false,`